icons/obj/station_map.dm

```diff
@@ -0,0 +1,89 @@
+///This is exclusively for the AI map right now
+/obj/station_map
+	name = "AI station map"
+	var/static/icon/map_icon
+
+	var/x_max = 1
+	var/x_min
+	var/y_max = 1
+	var/y_min
+
+	var/border_width = 20
+
+	New()
+		..()
+		x_min = world.maxx
+		y_min = world.maxy
+		if (!map_icon)
+			map_icon = icon("icons/obj/station_map.dmi", "blank")
+			render_map()
+			zoom_map()
+			pixel_y += 20 //magic numbers because ByondUI mis-aligns by a few pixels
+			pixel_x += 8
+		icon = map_icon
+
+	Click(location, control, params)
+		if (!isAIeye(usr)) //no humans teleporting
+			return
+		var/list/param_list = params2list(params)
+		if ("left" in param_list)
+			var/x = text2num(param_list["icon-x"])
+			var/y = text2num(param_list["icon-y"])
+			var/turf/clicked = locate(x, y, Z_LEVEL_STATION) //station z level macro
+			usr.loc = clicked
+		if ("right" in param_list)
+			return TRUE
+
+	//generates the map from the current station layout
+	proc/render_map()
+		for (var/y = world.maxy, y > 0, y--)
```

Comment:
Unsure if you're unaware of fast fors with a backwards step or you have a reason.
```suggestion
		for (var/y in world.maxy to 1 step -1)
```

Reply:
yep, did not know the step syntax (couldn't see it in the byond docs anywhere?)